patches/minecraft/net/minecraft/entity/item/EntityMinecart.java.patch

```diff
@@ -408,6 +453,39 @@
 +    }
 +
 +    /**
++     * Called in {@link #moveAlongTrack(BlockPos, IBlockState)}, this method allows
++     * modded minecarts to determine the rider's effects on the minecart.
++     *
++     * <p>See {@link #BOOSTING_MASK} and {@link #SLOWING_MASK} for details about flags.
++     *
++     * @param entity the rider, may be {@code null}
++     * @param flags the flags indicating the cart's status before the handling
++     * @return the updated flags
++     */
++    protected int handleRider(@Nullable Entity entity, int flags)
```

Comment:
Instead of moving/copying the vanilla code here, I think there should be a return value from here that means "do the vanilla logic". Mods that override this can do something else.

Reply:
good idea. then i can move the whole thing to an if block.